mesonbuild/interpreter.py

```diff
@@ -2512,7 +2512,14 @@ def add_build_def_file(self, f):
         elif os.path.isfile(f) and not f.startswith('/dev'):
             srcdir = Path(self.environment.get_source_dir())
             builddir = Path(self.environment.get_build_dir())
-            f = Path(f).resolve()
+            f = Path(f)
+            try:
+                f = Path(f).resolve()
+            except:
```

Comment:
It looks like from the docs this should either be `except FileNotFoundError` or, if we're hitting a really bad case `except (FileNotFoundError, RuntimeError)`